modules/core/04-channel/v2/keeper/msg_server.go

```diff
@@ -163,13 +163,21 @@ func (k *Keeper) RecvPacket(ctx context.Context, msg *types.MsgRecvPacket) (*typ
 		ack.AppAcknowledgements = append(ack.AppAcknowledgements, res.Acknowledgement)
 	}
 
+	if len(ack.AppAcknowledgements) != len(msg.Packet.Payloads) {
```

Comment:
moved here from internal func. its unreachable though? Not usre if this should stay here,